cmd/run.go

```diff
@@ -77,16 +74,8 @@ func bindRunFlags(cmd *cobra.Command, config *app.Config) {
 	cmd.Flags().BoolVar(&config.BuilderAPI, "builder-api", false, "Enables the builder api. Will only produce builder blocks. Builder API must also be enabled on the validator client. Beacon node must be connected to a builder-relay to access the builder network.")
 
 	wrapPreRunE(cmd, func(cmd *cobra.Command, args []string) error {
-		ctx := log.WithTopic(cmd.Context(), "cmd")
-		if beaconNodeAddr != "" {
-			log.Warn(ctx, "Deprecated flag 'beacon-node-endpoint' used, please use new flag 'beacon-node-endpoints'", nil)
-			config.BeaconNodeAddrs = []string{beaconNodeAddr}
-		} else if beaconNodeAddr != "" && len(config.BeaconNodeAddrs) > 0 {
-			log.Warn(ctx, "Deprecated flag 'beacon-node-endpoint' ignored since new flag 'beacon-node-endpoints' takes precedence",
-				nil, z.Str("beacon-node-endpoint", beaconNodeAddr), z.Any("beacon-node-endpoints", config.BeaconNodeAddrs))
-		} else if len(config.BeaconNodeAddrs) == 0 && !config.SimnetBMock {
+		if len(config.BeaconNodeAddrs) == 0 && !config.SimnetBMock {
 			return errors.New("either flag 'beacon-node-endpoints' or flag 'simnet-beacon-mock=true' must be specified")
-			// TODO(corver): Use MarkFlagsRequiredTogether once beacon-node-endpoint is removed.
```

Comment:
why not do this?

Reply:
`cmd.MarkFlagsRequiredTogether("beacon-node-endpoints", "simnet-beacon-mock")` will only error if both the flags are not provided together. However, i found out that `BeaconNodeAddrs` is evaluated to `[]` not nil, so the function doesn't panic since it understands that the flag WAS provided

Reply:
Also, explicitly returning the error is "more" clean as we can specify an error message